metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -1994,6 +2014,24 @@ private void listReassigningTopic(ListPartitionReassignmentsResponseData respons
             setReplicas(Replicas.toList(partition.replicas)));
     }
 
+    // Make it public to be visible in test.
+    public Integer getTopicEffectiveMinIsr(String topicName) {
+        Integer currentMinIsr = Integer.parseInt(configurationControl.getTopicConfigs(topicName)
+            .getOrDefault(MIN_IN_SYNC_REPLICAS_CONFIG, String.valueOf(defaultMinInSyncIsr)));
+        Integer replicationFactor = (int) defaultReplicationFactor;
+        try {
+            Uuid topicId = topicsByName.get(topicName);
+            replicationFactor = topics.get(topicId).parts.get(0).replicas.length;
+        } catch (Exception e) {
+            log.debug("Can't find the replication factor for topic: " + topicName);
+        }
```

Comment:
Let's not rely on an error here. We can check if the replicas array is zero-length